xarray/core/dataset.py

```diff
@@ -1231,11 +1273,15 @@ def isel(self, drop=False, **indexers):
             if not (drop and name in var_indexers):
                 variables[name] = new_var
 
-        # attach coordinate in indexers
-        variables.update(coord_vars)
-
-        coord_names = set(variables) & set(self._coord_names) | set(coord_vars)
-        return self._replace_vars_and_dims(variables, coord_names=coord_names)
+        coord_names = set(variables) & set(self._coord_names)
+        selected = self._replace_vars_and_dims(
+                        variables, coord_names=coord_names)
+        # Dataset consisting of only coordinate.
+        coord_vars = selected._get_indexers_coordinates(indexers)
+        coords = self._replace_vars_and_dims(
+                        coord_vars, coord_names=list(coord_vars.keys()))
+        # Drop conflicted variable from indexers coordinate
+        return coords.update(selected)
```

Comment:
Is there any existing logic for this?
What I want to do is 
+ to add coordinates in `coord` to `selected` if there is no conflict.
+ to silently ignore conflicted coordinates.

`coords.update(selected)` seems to do this, but I am not sure it is straightforward enough.